Include/internal/pycore_dict.h

```diff
@@ -264,6 +264,37 @@ _PyDictValues_AddToInsertionOrder(PyDictValues *values, Py_ssize_t ix)
     *size_ptr = size;
 }
 
+#ifdef Py_GIL_DISABLED
+
+static inline bool
+_PyDictOrValues_TryStoreValue(PyObject *obj, Py_ssize_t index, PyObject *value)
+{
+    bool success;
+    Py_BEGIN_CRITICAL_SECTION(obj);
+
+    PyDictOrValues *dorv_ptr = _PyObject_DictOrValuesPointer(obj);
+    if (!_PyDictOrValues_IsValues(*dorv_ptr)) {
+        success = false;
+        goto exit;
+    }
+
+    PyDictValues *values = _PyDictOrValues_GetValues(*dorv_ptr);
+    PyObject *old_value = _Py_atomic_load_ptr_relaxed(&values->values[index]);
+    _Py_atomic_store_ptr_relaxed(&values->values[index], value);
```

Comment:
Should be "release". For objects/pointers that are accessed without locks, I think we generally want at least "release" for stores to ensure that any writes in the pointed-to data are visible before the pointer.